File: 1431.kids-with-the-greatest-number-of-candies.java
/*
 * @lc app=leetcode id=1431 lang=java
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = Arrays.stream(candies).max().orElse(-1);
        List<Boolean> ret = new ArrayList<>();
        for (int ix = 0; ix != candies.length; ++ix) {
            ret.add(candies[ix] + extraCandies >= max);
        }
        return ret;
    }
}
// @lc code=end

File: 1431.kids-with-the-greatest-number-of-candies.cpp
/*
 * @lc app=leetcode id=1431 lang=cpp
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start

#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ret(candies.size(), false);
        const auto max_el = *max_element(candies.cbegin(), candies.cend());
        transform(candies.cbegin(), candies.cend(), ret.begin(), [&](const auto el) {
            return el + extraCandies >= max_el;
        });
        return ret;
    }
};
// @lc code=end
